presto-main/src/main/java/io/prestosql/security/StatementAccessControl.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.security;
+
+import io.prestosql.Session;
+import io.prestosql.spi.security.AccessDeniedException;
+import io.prestosql.sql.parser.SqlParser;
+
+public interface StatementAccessControl
```

Comment:
this should be part of `spi`

Reply:
Right. Instead of `Session` you could try passing `ConnectorSession` instead.
If you need `SqlParser` instance you should make your pluging dependent on `presto-main` and instantiate parser locally.

Reply:
`ConnectorSession` is connector related. I think `SecurityContext` (or `SecuritySession) should be used here, see: https://github.com/prestosql/presto/pull/171

Reply:
@sopel39 - **this should be part of spi**
The problem with plugin approach used for connectors in Presto is that one has to define SPI (service provider interface) in presto-spi code module. Our interface requires 'Session' object and that is there in presto-main. presto-main already depends on presto-spi and adding an interface to presto-spi with Session object is creating circular dependency

@sopel39 , @kokosing - ConnectorSession Vs Session
In Session, catalog and schema properties exist. If the table name is not fully qualified, then the following are used to construct the fully qualified table name.     
private final Optional<String> catalog;
private final Optional<String> schema;

The above properties are not there in ConnectorSession.

@sopel39, @kokosing  - SqlParser 
SqlParser is needed for parsing the query. Shall instantiate it in plugin

Reply:
> In Session, catalog and schema properties exist. If the table name is not fully qualified, then the following are used to construct the fully qualified table name.

@raviann if some fields are `ConnectorSession` or `SecurityContext` are missing then you could create your own interface (e.g: `StatementRewriteContext`) or extend `SecurityContext`.

> SqlParser is needed for parsing the query. Shall instantiate it in plugin

You can construct `SqlParser` directly in your plugin, but you need to have POM dependency on `presto-main`.

Reply:
Shall keep dependency on presto-main. That should be fine

Reply:
@sopel39 Shall create StatementRewriteContext to have the necessary properties